api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java

```diff
@@ -249,27 +219,47 @@ private PatientNameSearch getPatientNameSearch() {
 			        + "a bean implementing PatientNameSearch");
 		}
 	}
-	
-	/**
-	 * @see org.openmrs.api.GlobalPropertyListener#supportsPropertyName(java.lang.String)
-	 */
-	@Override
-	public boolean supportsPropertyName(String gpName) {
-		return RegistrationCoreConstants.GP_IDENTIFIER_SOURCE_ID.equals(gpName);
-	}
-	
+
 	@Override
 	public List<PatientAndMatchQuality> findFastSimilarPatients(Patient patient, Map<String, Object> otherDataPoints,
 	                                                            Double cutoff, Integer maxResults) {
-		return getFastSimilarPatientSearchAlgorithm().findSimilarPatients(patient, otherDataPoints, cutoff, maxResults);
+		List<PatientAndMatchQuality> matches = new LinkedList<PatientAndMatchQuality>();
+
+		List<PatientAndMatchQuality> localMatches = getFastSimilarPatientSearchAlgorithm()
+				.findSimilarPatients(patient, otherDataPoints, cutoff, maxResults);
+		matches.addAll(localMatches);
+
+		if (coreProperties.isMpiEnabled()) {
+			List<PatientAndMatchQuality> mpiMatches = coreProperties.getMpiProvider()
+					.findProbablySimilarPatients(patient, otherDataPoints, cutoff, maxResults);
```

Comment:
The name of the method we are calling here is confusing. "findProbablySimilarPatients" should instead be "findSimilarPatients". (They are _definitely_ similar, not probably similar.)